cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go

```diff
@@ -787,6 +803,22 @@ func (o *ScaleUpOrchestrator) ComputeSimilarNodeGroups(
 	return validSimilarNodeGroups
 }
 
+func (o *ScaleUpOrchestrator) newQuotasTracker() (*resourcequotas.Tracker, error) {
+	var nodes []*apiv1.Node
+	nodeInfos, err := o.autoscalingCtx.ClusterSnapshot.ListNodeInfos()
+	if err != nil {
+		return nil, err
+	}
+	for _, nodeInfo := range nodeInfos {
+		node := nodeInfo.Node()
+		if utils.IsVirtualKubeletNode(node) {
+			continue
+		}
+		nodes = append(nodes, nodeInfo.Node())
+	}
+	return o.quotasTrackerFactory.NewQuotasTracker(o.autoscalingCtx, nodes)
+}
```

Comment:
That probably could be done better:
1. Checking for virtual kubelet nodes could be done via `resourcequotas.NodeFilter`
2. Passing all the nodes, including upcoming nodes via `nodes` parameter in `ScaleUp` (this way we could also remove some logic in `ScaleUp` around upcoming nodes, which are used for checking total node limits)

That would make it also easier to integrate with the other implementations of the `Orchestrator`

Notes to interviewers: do you think it's worth the hassle?